tests/test_async_db_pool.py

```
@@ -0,0 +1,352 @@
+        """Test that connection rolls back on error."""
+        async with self.pool.get_connection() as conn:
+            await conn.execute(
+                "CREATE TABLE IF NOT EXISTS test_rollback (id INTEGER PRIMARY KEY, value TEXT)"
+            )
+            await conn.execute(
+                "INSERT INTO test_rollback (value) VALUES (?)", ("before_error",)
+            )
+            await conn.commit()
+
+            # Simulate an error
+            with self.assertRaises(Exception):
```

Ruff (0.14.3)

Do not assert blind exception: `Exception`

(B017)